examples/attempts_with_dlx_and_explicit_exceptions.rb

```diff
@@ -0,0 +1,73 @@
+# attempts_with_dead_letter_and_exponential_backoff.rb
+# ! check the examples/README.rdoc for information on starting your redis/rabbit !
+#
+# start it with ruby attempts_with_dead_letter_and_exponential_backoff.rb
+
+require "rubygems"
+require File.expand_path("../lib/beetle", File.dirname(__FILE__))
+
+require 'byebug'
+
+# set Beetle log level to info, less noisy than debug
+Beetle.config.logger.level = Logger::INFO
+
+# setup client with dead lettering enabled
+config = Beetle::Configuration.new
+config.dead_lettering_enabled = true
+config.dead_lettering_msg_ttl = 1000 # millis
+client = Beetle::Client.new(config)
+client.register_queue(:test)
+client.register_message(:test)
+
+# purge the test queue
+client.purge(:test)
+
+# empty the dedup store
+client.deduplication_store.flushdb
+
+# setup our counter
+$completed = 0
+$exceptions_limit = 4
+
+# store the start time
+$start_time = Time.now.to_f
+
+ReenqueueError = Class.new(StandardError)
+
+# declare a handler class for deferred/delayed message processing
+class Handler < Beetle::Handler
+  def process
+    logger.info message.data
+    $completed += 1
+
+    delay = (1..4).to_a.sample
+    message.instance_variable_set(:@delay, delay)
+    message.set_delay!
```

Comment:
this is obviously a hack to set a dynamic delay for the next run. At least i figured out that it works in general.

Reply:
but that worx client side (within beetle being triggered by the server).

real per message-ttl would work server side, as far as I understand the rabbitmq docs